test/e2e/test_helpers.go

```diff
@@ -0,0 +1,231 @@
+		InfrastructureProvider:   clusterctl.DefaultInfrastructureProvider,
+		Flavor:                   params.flavor,
+		Namespace:                params.namespace.Name,
+		ClusterName:              params.clusterName,
+		KubernetesVersion:        params.e2eConfig.GetVariable(KubernetesVersion),
+		ControlPlaneMachineCount: pointer.Int64Ptr(1),
+		WorkerMachineCount:       pointer.Int64Ptr(1),
+	}
+
+	createClusterFromConfig(ctx, clusterctl.ApplyClusterTemplateAndWaitInput{
+		ClusterProxy:                 params.bootstrapClusterProxy,
+		ConfigCluster:                cc,
+		WaitForClusterIntervals:      params.e2eConfig.GetIntervals("", "wait-cluster"),
+		WaitForControlPlaneIntervals: params.e2eConfig.GetIntervals("", "wait-control-plane"),
+		WaitForMachineDeployments:    params.e2eConfig.GetIntervals("", "wait-worker-nodes"),
+	}, clusterResources)
+}
+
+func generateTestClusterName(specName string) string {
+	return fmt.Sprintf("%s-%s", specName, util.RandomString(6))
+}
+
+func createClusterFromConfig(ctx context.Context, input clusterctl.ApplyClusterTemplateAndWaitInput, result *clusterctl.ApplyClusterTemplateAndWaitResult) {
+	Expect(ctx).NotTo(BeNil(), "ctx is required for createClusterFromConfig")
+	Expect(input.ClusterProxy).ToNot(BeNil(), "Invalid argument. input.ClusterProxy can't be nil when calling createClusterFromConfig")
+	Expect(result).ToNot(BeNil(), "Invalid argument. result can't be nil when calling createClusterFromConfig")
+	Expect(input.ConfigCluster.ControlPlaneMachineCount).ToNot(BeNil())
+	Expect(input.ConfigCluster.WorkerMachineCount).ToNot(BeNil())
+
+	clusterTemplate := clusterctl.ConfigCluster(ctx, clusterctl.ConfigClusterInput{
+		KubeconfigPath:           input.ConfigCluster.KubeconfigPath,
+		ClusterctlConfigPath:     input.ConfigCluster.ClusterctlConfigPath,
+		Flavor:                   input.ConfigCluster.Flavor,
+		Namespace:                input.ConfigCluster.Namespace,
+		ClusterName:              input.ConfigCluster.ClusterName,
+		KubernetesVersion:        input.ConfigCluster.KubernetesVersion,
+		ControlPlaneMachineCount: input.ConfigCluster.ControlPlaneMachineCount,
+		WorkerMachineCount:       input.ConfigCluster.WorkerMachineCount,
+		InfrastructureProvider:   input.ConfigCluster.InfrastructureProvider,
+		LogFolder:                input.ConfigCluster.LogFolder,
+	})
+	Expect(clusterTemplate).ToNot(BeNil())
+
+	Expect(input.ClusterProxy.Apply(ctx, clusterTemplate, input.Args...)).To(Succeed())
+
+	result.Cluster = framework.GetClusterByName(ctx, framework.GetClusterByNameInput{
+		Getter:    input.ClusterProxy.GetClient(),
+		Name:      input.ConfigCluster.ClusterName,
+		Namespace: input.ConfigCluster.Namespace,
+	})
+	Expect(result.Cluster).ToNot(BeNil())
+}
+
+type getNutanixClusterByNameInput struct {
+	Getter    framework.Getter
+	Name      string
+	Namespace string
+}
+
+func getNutanixClusterByName(ctx context.Context, input getNutanixClusterByNameInput) *infrav1.NutanixCluster {
```

Comment:
can we make these gloval functions, a member of an interface  so that it will be easy to understand what all things we already have.

Reply:
interfaces could distinguish between, clusterctl wrappers vs test_infra wrappers vs api server wrappers vs nutanix specific wrappers.